include/OledDisplay.h

```diff
@@ -0,0 +1,21 @@
+#ifndef OLEDDISPLAY_H
+#define OLEDDISPLAY_H
+
+#include <U8g2lib.h>
+
+#define SCL 22
+#define SDA 21
+
+class Display {
+  private:
+    U8G2_PCD8544_84X48_1_4W_HW_SPI _nokia = U8G2_PCD8544_84X48_1_4W_HW_SPI(U8G2_R0, 5, 4, 16);
```

Comment:
Ich würde hier, um RAM zu sparen erst einmal nur pointer auf die jeweiligen klassen anlegen. Wenn das entsprechende display gewählt wird kann man die instanz immer noch  erzeugen.
Generell könnte der Pointer auch auf ein U8G2 object zeigen. Dann könntest du in der cpp datei die ganzen verschiedenen switch/case aufrufe sparen und einfach auf die methoden der übergeordneten klasse zugreifen.